dao/FashionKnex.js

```diff
@@ -80,29 +84,74 @@ class FashionKnex {
       })
   }
 
+  /**
+  * Find a Entity by `id` in the database
+  *
+  * @param id {String} - Id to search for
+  * @param options {Object}
+  *   options.select {String|Array} - Properties to select in the query
+  */
   async findById (id, {select = '*'} = {}) {
     const tableName = this._tableName
 
-    let wrapped
     let result
 
+    if (typeof select === 'string' || typeof select === 'undefined') {
+      result = this.select(select)
+    } else if (Array.isArray(select)) {
+      result = this.select(...select)
+    } else {
+      throw new Error(`Invalid select value "${select}". The select option ` +
+        `should either be a string or an array.`)
+    }
+
     try {
-      result = await this._knex
-        .select(select)
-        .from(tableName)
+      result = await result
         .where('id', id)
         .limit(1)
     } catch (err) {
       this._logger.error(`Error fetching id "${id}" from  table "${tableName}"`, err)
       throw err
     }
 
-    let errors = []
+    return this.wrap(result[0])
+  }
 
-    wrapped = this._modelType.wrap(result[0], errors)
+  /**
+  * Select query builder that selects from our current table. Uses `arguments`
+  * to determine what is selected.
+  *
+  * ex.
+  *
+  * dao.select() -> select * from tableName
+  * dao.select('name', 'type') -> select "name", "type" from tableName
+  */
+  select () {
+    const tableName = this._tableName
+    let result
+
+    if (!arguments.length) {
+      result = this._knex.select('*')
+    } else {
+      let select = []
+      for (let k in arguments) {
+        select.push(arguments[k])
+      }
+      result = this._knex.select(...select)
+    }
+
+    return result.from(tableName)
```

Comment:
There's a lot of extra logic in this function. Since `arguments` is iterable, you should be able to skip the `select` building entirely and this entire function simply becomes one line: `return this._knex.select(...arguments).from(this._tableName)`

Reply:
The reason I left this off was because object spread isn't fully supported throughout node 8, but doesn't really matter since we are committed to more recent versions.

Reply:
In case you didn't know, it's supported in Node 8.3+ without a flag. Node 8.2.1 with a flag.